### DISCUSSION TITLE
defineProps<{ msg: string }>()   Parsing error: Unexpected token )

### DISCUSSION THREAD

**Discussion by yf339339** [eslint/eslint#17221] [Q&A]
### Environment

Node version: 16.20.0
npm version:  8.19.4
Local ESLint version:  8.41.0
Global ESLint version:  no
Operating System:  win 10

### What parser are you using?

Default (Espree)

### What did you do?

vue3 default file  HelloWorld.vue error
![image](https://github.com/eslint/eslint/assets/28554074/e6a1d7a8-b1c8-43ca-aab8-caf679b713ac)
![image](https://github.com/eslint/eslint/assets/28554074/67316a31-848b-45cc-9085-ed7772a998a6)

### What did you expect to happen?

defineProps<{ msg: string }>()  no error

### What actually happened?

vue3 default file  HelloWorld.vue error
![image](https://github.com/eslint/eslint/assets/28554074/e6a1d7a8-b1c8-43ca-aab8-caf679b713ac)
![image](https://github.com/eslint/eslint/assets/28554074/67316a31-848b-45cc-9085-ed7772a998a6)

### Link to Minimal Reproducible Example

no

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
not very familiar with the vue+ts integration, but it's mostly a configure issue. converting to a discussion.

**Comment by daiwanxing** [comment]
Hi @yf339339, please checkout whether installed [`eslint-plugin-vue`](https://eslint.vuejs.org/user-guide/). If not, you should install it.

an example within use `eslint-plugin-vue` in eslint

```js
// .eslintrc.js
module.exports = {
 // .....
 parser: "vue-eslint-parser",
 extends: ["plugin:vue/vue3-recommended"],
}
```
I hope this will resolve your problem.

**Reply by yf339339** [reply]
Thanks，I installed it，I built the vue project according to the official vite documentation, and then installed eslint separately

**Comment by aladdin-add** [comment]
to use ts-eslint, can you try the `parserOptions.parser: "@typescript-eslint/parser"`?

https://eslint.vuejs.org/user-guide/#how-to-use-a-custom-parser


and were you using `eslint init @eslint/config` to create the config. I think we might need to fix it in `@eslint/create-config` if so.

**Comment by aladdin-add** [comment]
I was able to repro it as followings:

![image](https://github.com/eslint/eslint/assets/13050025/3859989d-e24d-4459-a0f9-6a316933ded3)

I think you can just change reorder the extends:
```js
extends: ["standard-with-typescript", "plugin:vue/vue3-essential"]
```
and remove the `"parser": "@typescript-eslint/parser"`.

thanks for the report, will make a fix later. :)

**Reply by yf339339** [reply]
Thanks, but it doesn't seem to work，I built the vue project according to the official vite documentation, and then installed eslint separately
![1685416702496](https://github.com/eslint/eslint/assets/28554074/44a0acfd-50f4-4675-a18f-80c92960d51f)
![1685415891375](https://github.com/eslint/eslint/assets/28554074/c04d2c43-8344-4527-80fb-d3c278da7b1d)

**Reply by aladdin-add** [reply]
![image](https://github.com/eslint/eslint/assets/13050025/adf2ae0b-a8d2-4d0b-83f7-0ec920ec0468)

try this.

**Reply by yf339339** [reply]
Thanks,It's ok

**Comment by shenhao98** [comment]
老哥 你这个问题解决没有 我也遇到了相同的问题

**Reply by yf339339** [reply]
按上面那个人说的操作解决了
![image](https://github.com/eslint/eslint/assets/28554074/de67b62c-c545-4bf5-bf73-79e10437f708)

**Comment by lewebsimple** [comment]
Fixed by installing typescript: `pnpm add -D typescript`
